trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/signing/InternalSigningController.java

```diff
@@ -100,6 +100,7 @@ public String signRequest(
                 metadata,
                 region,
                 requestDate,
+                Optional.empty(),
```

Comment:
Please pull up the expiry into the interface/base. We will need to pre-sign requests for external consumption (as mentioned in Slack).

Reply:
I've refactored the API a bit so that we compute the expiry at the start and treat the `RequestAuthorization` as invalid immediately if it has expired.

I have modified the signing API so it returns a `RequestAuthorization` which (unlike the String it returned before) should be usable to compute presigned URLs too